clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java

```diff
@@ -573,10 +573,13 @@ public void tick() throws Exception {
     }
 
     private void updateMasterClusterSyncMetrics() {
-        var stats = stateVersionTracker.getAggregatedClusterStats().getAggregatedStats();
-        if (stats.hasUpdatesFromAllDistributors()) {
-            GlobalBucketSyncStatsCalculator.clusterBucketsOutOfSyncRatio(stats.getGlobalStats())
+        var stats = stateVersionTracker.getAggregatedClusterStats();
+        var aggrStats = stats.getAggregatedStats();
+        if (aggrStats.hasUpdatesFromAllDistributors()) {
```

Comment:
If we go to the `else`-part for an extended period of time, the stored-document-count metrics may appear to be stable even though they are stale and not trustworthy.  What's the best way to capture this?  Have another metric on the ratio of distributors that we have updates from, and... alert on it if it stays <1.0 for too long?  Or always show such a graph next to stored-document-count metrics?  Set the metric to 0 if the `if` has not been entered within the last X minutes?

Reply:
We have a dedicated, time-sensitive `cluster-controller.nodes-not-converged` metric that is intended to capture this scenario, and that can generate actionable alerts if > 0. I therefore think it should be OK to not have any explicit failure detection as part of these new metrics.